apps/src/bin/anoma-node/cli.rs

```diff
@@ -6,6 +6,7 @@ use eyre::{Context, Result};
 
 pub fn main() -> Result<()> {
     let (cmd, mut ctx) = cli::anoma_node_cli();
+    ctx.config.ledger.tendermint.tendermint_mode = ctx.global_args.mode.clone();
```

Comment:
I think maybe this would be better in somewhere in the `cli::Context::new`. The `global_args.mode` defaults to Validator when it's not specified though, so it looks like this will always override it to be Validator?

Reply:
I thought that some of the e2e tests specificied full node. Otherwise they would fail.

Reply:
Ah right, what I was thinking about is if you set it in your config, but you don't set the global arg, it will get overridden on start-up to the default

Reply:
it might be better to make the global arg optional and only override it if it's set